test.c

```diff
@@ -219,16 +220,60 @@ void create_objects(void)
     /* cJSON_ReplaceItemInObject(cJSON_GetArrayItem(root, 1), "City", cJSON_CreateIntArray(ids, 4)); */
 
     out = cJSON_Print(root);
-    cJSON_Delete(root);
     printf("%s\n", out);
+
+    printf("Test cJSON_PrintPreallocated:\n");
+    /* create buffer */
+    size_t len = strlen(out) + 1;
```

Comment:
It is not allowed to mix code and declarations in C89. I have a bug in the Makefile and CMakelists.txt so that the compiler option is not enforced, sorry for that.

I suggest you create a new function just for testing preallocated printing and call it from main (you can pass one of the texts to it).